### PR TITLE
Use the VM's `precompiler2` tool instead of `dart`

### PR DESCRIPTION
**Pull request by kmillikin**
It's not necessary to build the entire SDK, the VM's `precompiler2` tool can produce AOT snapshots.

### CODE REVIEW

**Review by felangel — APPROVED** [review]

File: bin/run_ffi_tests.dart [hunk 1]
```diff
@@ -134,14 +134,15 @@ final tests = [
...
 void main(List<String> args) {
-  final engineDir = '../dart-sdk/sdk';
-  final compilerBuildDir = 'xcodebuild/DebugARM64';
-  final runtimeBuildDir = 'xcodebuild/DebugSIMARM64';
-  final runtimeBuildDirPath = p.absolute('$engineDir/$runtimeBuildDir');
-  final compilerBuildDirPath = p.absolute('$engineDir/$compilerBuildDir');
+  // Relative to this script's working directory.
+  final sdkDir = '../dart-sdk/sdk';
+  final compilerConf = 'DebugARM64';
+  final runtimeConf = 'DebugSIMARM64';
```
Comment (bryanoltman) [inline bin/run_ffi_tests.dart]: These could be `const`

File: bin/run_ffi_tests.dart [hunk 2]
```diff
@@ -134,14 +134,15 @@ final tests = [
...
+  final buildDirPath = p.normalize('$sdkDir/xcodebuild/$runtimeConf');
 
-  final compilerPath = '$compilerBuildDirPath/dart-sdk/bin/dart';
-  final runtimePath = '$runtimeBuildDirPath/dart_precompiled_runtime_product';
+  // Relative to buildDirPath.
+  final compilerPath = p.normalize('../../pkg/vm/tool/precompiler2');
+  final runtimePath = './dart_precompiled_runtime_product';
```
Comment (bryanoltman) [inline bin/run_ffi_tests.dart]: `const` here also

**Review by bryanoltman — APPROVED** [review]

File: bin/run_ffi_tests.dart (hunk 1 above)
Comment (felangel) [reply bin/run_ffi_tests.dart]: Good call. For consistency we could use the same analysis_options as in `shorebird_cli` and other packages

Comment (kmillikin) [reply bin/run_ffi_tests.dart]: Done, thanks for the suggestion.

**Review by eseidel — COMMENTED** [review]
This is gorgeous, thanks!

**Pull request merged**